src/main/java/frc/robot/subsystems/Feeder.java

```diff
@@ -47,13 +63,19 @@ public ColorValue getColor() {
     }
 
     public boolean pieceSeen(boolean incoming) {
-        ColorMatchResult latestResult = colorSensor.getMatchedColor();
-        double confidence = incoming ? Constants.COLOR_CONFIDENCE_RATE_INCOMING : Constants.COLOR_CONFIDENCE_RATE_BACKDRIVE;
-        return latestResult.confidence > confidence;
+        for (ColorMatchResult match: resultsSinceLastTic){
+            double confidence = incoming ? Constants.COLOR_CONFIDENCE_RATE_INCOMING : Constants.COLOR_CONFIDENCE_RATE_BACKDRIVE;
+            if (match.confidence > confidence){
+                return true;
+            }
+        }
+        return false;
     }
 
     @Override
     public void periodic() {
+        resultsSinceLastTic.clear();
+        colorReadingBuffer.drainTo(resultsSinceLastTic);
```

Comment:
This assumes that the periodic and pieceSeen run in the same thread (otherwise you'll get a concurrentModificationException) which may be correct assumption but still...

Reply:
I set variable to an atomic integer. 
One thing to keep in mind if we run into problems is that there is a period between the subsystem periodic (called first) and any commands isFinished calls that read the color sensor. So if let's say the piece enters the sensor but feeder periodic has already said that the piece is not there, then the motors would not stop for another tic. I don't think this will be noticeable (1/50 of a second delay should be fine) but I thought I would bring it up.